---

# **Submission Processing** <br>

This notebook processes the submission for the 'askscience' subreddit. It then also processes the comments of the 'askscience' subreddit and combines them into a single file.<br>

**Part 1:** Read the askscience submissions.<br>
**Part 2:** Read the askscience comments.<br>
**Part 3:** Combine outputs from Part 1 and Part 2 and process the file, making it ready to perform node embeddings and word embeddings.

---

Outputfile:<br>
1. 'data_merged_askscience.csv': the data containing both submissions and comments and processed, ready to be analyzed.

.

.


---
# **Part 1: Read the askscience submissions.**

In this section, I have read the ask science subreddit submissions. Also the data is converted to a dataframe.

---
.

Check if cuda is being used

In [ ]:
import torch
if torch.cuda.is_available():
    device_name = torch.device("cuda")
else:
    device_name = torch.device('cpu')
print("Using {}.".format(device_name))

Using cpu.


Connect to drive

In [ ]:
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


Read the data containing askscience subreddit submissions

In [ ]:
import pandas as pd
data_submissions = pd.read_json('/content/gdrive/MyDrive/Colab Notebooks/askscience/askscience_submissions_10_2022.ndjson.ndjson')

ValueError: ignored

throws error which means that the file has to be processed

The first step is to convert the file into a valid json format. For this, place the entire file contents in two square brackets. Also, seperate each of the rows which are enclosed in two curly braces with a comma. First we will parse through the entire file and add a comma after every row/curly bracket pair

In [ ]:
b = open("/content/gdrive/MyDrive/Colab Notebooks/askscience/askscience_submissions_10_2022.ndjson","r")
bb = b.readlines()
index = 0

for x in bb:
  g = bb[index]+','
  print(g)
  with open('/content/gdrive/MyDrive/Colab Notebooks/askscience/submissions_askcience_processed.ndjson','a') as fle:
    fle.write(g)
  index += 1

Streaming output truncated to the last 5000 lines.
{"id":"y8tesp","subreddit":"askscience","title":"Strangely specific question, but I need the answer.","author":"cthulhu4150","permalink":"/r/askscience/comments/y8tesp/strangely_specific_question_but_i_need_the_answer/","num_comments":1,"url":"https://www.reddit.com/r/askscience/comments/y8tesp/strangely_specific_question_but_i_need_the_answer/","score":1,"gilded":0,"created_utc":1666257891,"self_text":"[removed]","is_self":true,"retrieved_on":1668028361}
,
{"id":"y8tfyh","subreddit":"askscience","title":"Why is boredom so exhausting?","author":"boomer_wife","permalink":"/r/askscience/comments/y8tfyh/why_is_boredom_so_exhausting/","num_comments":0,"url":"https://www.reddit.com/r/askscience/comments/y8tfyh/why_is_boredom_so_exhausting/","score":1,"gilded":0,"created_utc":1666258007,"self_text":"[removed]","is_self":true,"retrieved_on":1668028359}
,
{"id":"y8q1ku","subreddit":"askscience","title":"Why are animal’s pupils so big?!","autho

But now the file has to be editted as there is an extra comma after the end of the last row. So download the file from drive, open it up and remove the extra comma and also add a square bracket at the beginning and at the end of the file, and upload it back to the drive

now the file 'submissions_askcience_processed.ndjson' has to be cleaned

In [ ]:
with open('/content/gdrive/MyDrive/Colab Notebooks/askscience/submissions_askcience_processed.ndjson','r') as fle1:
  cur_line = fle1.readlines()

output= ""
count = 0
for line in cur_line:
  line = line.replace("\\n", "" )
  line = line.replace('\\"', "" )
  line = line.replace('\\', "" )
  line = line.replace("'", "" )
  # line = line.replace('"', "" )
  if count==4:
    print(line)
  output+=line
  count+=1

with open('/content/gdrive/MyDrive/Colab Notebooks/askscience/submissions_askcience_processed_1.ndjson','w') as f2:
  f2.writelines(output)

,{"id":"xsofdj","subreddit":"askscience","title":"would we theoretically be able to see earths past?","author":"soggy-tampon","permalink":"/r/askscience/comments/xsofdj/would_we_theoretically_be_able_to_see_earths_past/","num_comments":1,"url":"https://www.reddit.com/r/askscience/comments/xsofdj/would_we_theoretically_be_able_to_see_earths_past/","score":1,"gilded":0,"created_utc":1664606931,"self_text":"[removed]","is_self":true,"retrieved_on":1668049257}



In [ ]:
import pandas as pd
data_submissions = pd.read_json('/content/gdrive/MyDrive/Colab Notebooks/askscience/submissions_askcience_processed_1.ndjson',encoding_errors='ignore')

In [ ]:
data_submissions.head(10)

id   subreddit                                              title  \
0  xsjqzy  askscience  Why do I poop after a glass or two of beer or ...   
1  xsju1p  askscience  If you heated water under immense pressure so ...   
2  xsjx0l  askscience  What is the evolutionary goal of nose elongati...   
3  xsk0ck  askscience  Does resting in bed count towards a good night...   
4  xsofdj  askscience  would we theoretically be able to see earths p...   
5  xsk9nb  askscience  How is the Calorie count of non-food items, su...   
6  xsotzf  askscience  Does lava loose temperature during it flow? If...   
7  xskc9v  askscience  I have often observed that human head movement...   
8  xsox5d  askscience         Why do EVs use a bunch of small batteries?   
9  xsoxf5  askscience  Will solar panels wrapped around a tower gener...   

                author                                          permalink  \
0        depressedchiq  /r/askscience/comments/xsjqzy/why_do_i_poop_af...   
1         zhongliabuse  /r/askscience/comments/xsju1p/if_you_heated_wa...   
2               _ozeki  /r/askscience/comments/xsjx0l/what_is_the_evol...   
3            [deleted]  /r/askscience/comments/xsk0ck/does_resting_in_...   
4         soggy-tampon  /r/askscience/comments/xsofdj/would_we_theoret...   
5           liudinglue  /r/askscience/comments/xsk9nb/how_is_the_calor...   
6  giftsignificant5286  /r/askscience/comments/xsotzf/does_lava_loose_...   
7            [deleted]  /r/askscience/comments/xskc9v/i_have_often_obs...   
8            kenwillis  /r/askscience/comments/xsox5d/why_do_evs_use_a...   
9              phaiyez  /r/askscience/comments/xsoxf5/will_solar_panel...   

   num_comments                                                url  score  \
0             0  https://www.reddit.com/r/askscience/comments/x...      1   
1             1  https://www.reddit.com/r/askscience/comments/x...      1   
2             0  https://www.reddit.com/r/askscience/comments/x...      1   
3             0                                                         1   
4             1  https://www.reddit.com/r/askscience/comments/x...      1   
5             0  https://www.reddit.com/r/askscience/comments/x...      1   
6             0  https://www.reddit.com/r/askscience/comments/x...      1   
7             0                                                         1   
8             0  https://www.reddit.com/r/askscience/comments/x...      1   
9             1  https://www.reddit.com/r/askscience/comments/x...      1   

   gilded  created_utc  self_text  is_self  retrieved_on  
0       0   1664591533  [removed]     True    1665426496  
1       0   1664591793  [removed]     True    1665426492  
2       0   1664592050  [removed]     True    1665426489  
3       0   1664592331  [removed]     True    1665426484  
4       0   1664606931  [removed]     True    1668049257  
5       0   1664593120  [removed]     True    1665426473  
6       0   1664608438  [removed]     True    1668049244  
7       0   1664593321  [removed]     True    1665426470  
8       0   1664608764  [removed]     True    1668049241  
9       0   1664608793  [removed]     True    1668049241

In [ ]:
print(len(data_submissions)) #length of data = 6835
print(len(pd.unique(data_submissions['subreddit']))) #number of subreddits considered = 1
print(len(pd.unique(data_submissions['id']))) #unique number of comments = 6835 #the data is at the submission level and each row is one submission
print(len(pd.unique(data_submissions['id'])))
print(len(pd.unique(data_submissions['author'])))

6835
1
6835
6835
4614


checking some stats from the submissions

In [ ]:
data_submissions.loc[data_submissions.author == '[deleted]', 'author'].count()

1156

In [ ]:
data_submissions.loc[data_submissions.author == '[removed]', 'author'].count()

0

In [ ]:
data_submissions.loc[data_submissions.self_text == '[deleted]', 'self_text'].count()

26

In [ ]:
data_submissions.loc[data_submissions.self_text == '[removed]', 'self_text'].count()

5141

In [ ]:
data_submissions.loc[data_submissions.title == '[deleted]', 'title'].count()

0

In [ ]:
data_submissions.loc[data_submissions.title == '[removed]', 'title'].count()

0

1156+26+5141 = 6323 which is the total number of comments.

printing a few titles to compare

In [ ]:
data_submissions['title'][700]

'[Serious] I understand that a butterfly goes through the cycle of being an egg, then a larva, pupa, and finally a butterfly. If we compare that to the life of a human, we would have a human birth equate to a butterfly emerging. If that is so, what stage would pregnancy be: larva or pupa?'

In [ ]:
data_submissions['title'][3000]

'I die and my body is put in escape velocity from the solar system to eternally float through space. How does decay effect it after a month, a thousand, a billion years?'

In [ ]:
data_submissions['title'][340]

'How long does the virus live/stay in the myocardium?'

assumption: an assumption that I am making is that the submission title will be used as the body

---
# **Part 2: Read the askscience comments.**

In this section, I have read the ask science subreddit. Also the data is converted to a dataframe.

---
.

In [ ]:
import pandas as pd
data_df = pd.read_json('/content/gdrive/MyDrive/Colab Notebooks/askscience/askscience_comments_10_2022.ndjson')

ValueError: ignored

throws error which means that the file has to be processed

The first step is to convert the file into a valid json format. For this, place the entire file contents in two square brackets. Also, seperate each of the rows which are enclosed in two curly braces with a comma. First we will parse through the entire file and add a comma after every row/curly bracket pair

In [ ]:
b = open("/content/gdrive/MyDrive/Colab Notebooks/askscience/askscience_comments_10_2022.ndjson","r")
bb = b.readlines()
index = 0

for x in bb:
  g = bb[index]+','
  print(g)
  with open('/content/gdrive/MyDrive/Colab Notebooks/askscience/askcience_processed.ndjson','a') as fle:
    fle.write(g)
  index += 1

Streaming output truncated to the last 5000 lines.
{"id":"iu0xdnj","subreddit":"askscience","body":"Hey I'm the same. Back to ED after not being in a clinical position for a long time, only this time I'm in a peds ED and it's really bad. RSV all day every day. We have so many kids that they don't even have private rooms for all of them so they go in shared open bays. Long wait times. Running out of high flow machines.","author":"evdczar","score":6,"gilded":0,"created_utc":1666897379,"parent_id":"iu0sbxi","link_id":"yere7r","retrieved_on":1667876590,"controversiality":0,"is_submitter":false}
,
{"id":"iu0xig7","subreddit":"askscience","body":"[removed]","author":"[deleted]","score":1,"gilded":0,"created_utc":1666897432,"parent_id":"yere7r","link_id":"yere7r","retrieved_on":1667876584,"controversiality":0,"is_submitter":false}
,
{"id":"iubg3ey","subreddit":"askscience","body":"Keep in mind that doctors are still human, and humans are susceptible to biases, misinformation, and politics. [A

But now the file has to be editted as there is an extra comma after the end of the last row. So download the file from drive, open it up and remove the extra comma and also add a square bracket at the beginning and at the end of the file, and upload it back to the drive

now the file 'askcience_processed.ndjson' has to be cleaned

In [ ]:
with open('/content/gdrive/MyDrive/Colab Notebooks/askscience/askcience_processed.ndjson','r') as fle1:
  cur_line = fle1.readlines()

#line 26606 has a problem
j = 0

output= ""
count = 0
for line in cur_line:
  j += 1
  if j == 26606:
    continue
  line = line.replace("\\n", "" )
  line = line.replace('\\"', "" )
  line = line.replace('\\', "" )
  line = line.replace("'", "" )
  # line = line.replace('"', "" )
  if count==4:
    print(line)
  output+=line
  count+=1

with open('/content/gdrive/MyDrive/Colab Notebooks/askscience/askcience_processed_1.ndjson','w') as f2:
  f2.writelines(output)

,{"id":"iqkfmj9","subreddit":"askscience","body":"It *absolutely* implies an expectation, even if it doesnt imply an inevitability. If you said He hasnt arrived yet, that means the person is *supposed* to arrive even if they turn out not to. You could stick with we dont know, but I say it just is because when it comes to things like fundamental properties of nature, I dont think the *causes* of such things are objects of science. If you think they are potential objects of science, Im curious about your perspective.How do you think youre going to determine *why* the speed of light is what is it, beyond *what* it is? Thats a question that invokes the cause of the universe itself. What kind of observation that is possible from within the universe could even *conceivably* answer that question? As far as science is concerned, reality is a closed system.","author":"chop1n","score":3,"gilded":0,"created_utc":1664583378,"parent_id":"iqker6l","link_id":"xs73nx","retrieved_on":1664960507,"contro

In [ ]:
import pandas as pd
data = pd.read_json('/content/gdrive/MyDrive/Colab Notebooks/askscience/askcience_processed_1.ndjson',encoding_errors='ignore')

In [ ]:
data.head(3)

id   subreddit                                               body  \
0  iqker6l  askscience  No it does not imply that. “We don’t yet know”...   
1  iqkewq0  askscience  while insect muscle might be similar to ours s...   
2  iqkfdmz  askscience                                          [removed]   

                author  score  gilded  created_utc parent_id link_id  \
0          omniskeptic      2       0   1664582942   iqkee0k  xs73nx   
1  regular_modern_girl    452       0   1664583016   iqjssf5  xs9pjy   
2            [deleted]      1       0   1664583252   iqkb49u  xs9pjy   

   retrieved_on  controversiality  is_submitter  
0    1664960533                 0         False  
1    1664960528                 0         False  
2    1664960514                 0         False

store this dataframe in a csv so that next time the code can be run from below by directly reading the processed cvs

In [ ]:
print(len(data)) #length of data = 26605
print(len(pd.unique(data['subreddit']))) #number of subreddits considered = 1
print(len(pd.unique(data['id']))) #unique number of comments = 26605 #the data is at the comment level
print(len(pd.unique(data['parent_id']))) #number of parent nodes = 10538
print(len(pd.unique(data['link_id']))) #number of submissions = 3004
print(len(pd.unique(data['author']))) #number of submissions = 6629
print(len(data.columns)) # = 12

26605
1
26605
10538
3004
6629
12


checking some stats from the comments

In [ ]:
data.loc[data.author == '[deleted]', 'author'].count()

13334

In [ ]:
data.loc[data.body == '[deleted]', 'body'].count()

369

In [ ]:
data.loc[data.body == '[removed]', 'body'].count()

12926

In [ ]:
data.loc[data.author == '[removed]', 'author'].count()

0

Confirming that all the rows from comments have this column value as 'False. when merging the comments with submissions, we can make this value 'True'

In [ ]:
print(len(data['is_submitter'] == 'False'))

26605


---
# **Part 3: Combine outputs from Part 1 and Part 2 and process the file, making it ready to perform node embeddings and word embeddings.**

---
.

In [ ]:
data_submissions.columns

Index(['id', 'subreddit', 'title', 'author', 'permalink', 'num_comments',
       'url', 'score', 'gilded', 'created_utc', 'self_text', 'is_self',
       'retrieved_on'],
      dtype='object')

In [ ]:
data.columns

Index(['id', 'subreddit', 'body', 'author', 'score', 'gilded', 'created_utc',
       'parent_id', 'link_id', 'retrieved_on', 'controversiality',
       'is_submitter'],
      dtype='object')

common columns are: id, subreddit, title or body, author, score, gilded, created_utc, retreived_on<br>

different columns:<br>
data_submissions: (13) permalink, num_comments, url, self_text, is_self<br>
data: (12) parent_id, link_id, controversiality, is_submitter

total number of columns = 17

In an attempt to merge the two dataframe, I will ensure that parent for all comments (except submissiona are found. Previously we only used comments for which there existed a parent_comment, but now by including submissions we also use comments whose parent is a submission itself.

In [ ]:
data_final = pd.DataFrame(columns=['id', 'subreddit', 'body', 'author', 'score', 'gilded', 'created_utc', 'retreived_on','permalink', 'num_comments', 'url', 'self_text', 'is_self', 'parent_id', 'link_id', 'controversiality', 'is_submitter' ])
data_final

Empty DataFrame
Columns: [id, subreddit, body, author, score, gilded, created_utc, retreived_on, permalink, num_comments, url, self_text, is_self, parent_id, link_id, controversiality, is_submitter]
Index: []

In [ ]:
def merge_submissions_comments(submissions_data, comments_data, df):

  print('processing the submissions data')
  j = 0
  for ind1, row1 in submissions_data.iterrows():
    j += 1
    if j % 1000 == 0:
      print('finished comment '+str(j)+'/'+str(len(submissions_data)))

    submission_id = row1['id']
    submission_subreddit = row1['subreddit']
    submission_body = row1['title']
    submission_author = row1['author']
    submission_score = row1['score']
    submission_gilded = row1['gilded']
    submission_created_utc = row1['created_utc']
    submission_retrieved_on = row1['retrieved_on']
    submission_permalink = row1['permalink']
    submission_num_comments = row1['num_comments']
    submission_url = row1['url']
    submission_self_text = row1['self_text']
    submission_is_self = row1['is_self']

    df.loc[len(df.index)] = [submission_id, submission_subreddit,submission_body, submission_author, submission_score,submission_gilded, submission_created_utc,
                             submission_retrieved_on, submission_permalink, submission_num_comments,submission_url, submission_self_text, submission_is_self,
                             '000000', '000000', 'novalue', True]

  print('processing the comments data')
  j = 0
  for ind2, row2 in comments_data.iterrows():
    j += 1
    if j % 1000 == 0:
      print('finished comment '+str(j)+'/'+str(len(comments_data)))

    comment_id = row2['id']
    comment_subreddit = row2['subreddit']
    comment_body = row2['body']
    comment_author = row2['author']
    comment_score = row2['score']
    comment_gilded = row2['gilded']
    comment_created_utc = row2['created_utc']
    comment_retrieved_on = row2['retrieved_on']
    comment_parent_id = row2['parent_id']
    comment_link_id = row2['link_id']
    comment_controversiality = row2['controversiality']
    comment_is_submitter = row2['is_submitter']


    df.loc[len(df.index)] = [comment_id, comment_subreddit,comment_body, comment_author, comment_score,comment_gilded, comment_created_utc,
                             comment_retrieved_on, '000000', '000000','000000', '000000', '000000',
                             comment_parent_id, comment_link_id, comment_controversiality, comment_is_submitter]

  return df


In [ ]:
data_final = merge_submissions_comments(data_submissions,data,data_final)
print(len(data_final)) #length of data
print(len(pd.unique(data_final['subreddit']))) #number of subreddits considered =
print(len(pd.unique(data_final['id']))) #unique number of comments = , the data is at the comment level =
print(len(pd.unique(data_final['author']))) #number of author =
print(len(pd.unique(data_final['parent_id']))) #number of parent nodes =
print(len(pd.unique(data_final['link_id']))) #number of submissions =
print(len(data_final.columns))

processing the submissions data
finished comment 1000/6835
finished comment 2000/6835
finished comment 3000/6835
finished comment 4000/6835
finished comment 5000/6835
finished comment 6000/6835
processing the comments data
finished comment 1000/26605
finished comment 2000/26605
finished comment 3000/26605
finished comment 4000/26605
finished comment 5000/26605
finished comment 6000/26605
finished comment 7000/26605
finished comment 8000/26605
finished comment 9000/26605
finished comment 10000/26605
finished comment 11000/26605
finished comment 12000/26605
finished comment 13000/26605
finished comment 14000/26605
finished comment 15000/26605
finished comment 16000/26605
finished comment 17000/26605
finished comment 18000/26605
finished comment 19000/26605
finished comment 20000/26605
finished comment 21000/26605
finished comment 22000/26605
finished comment 23000/26605
finished comment 24000/26605
finished comment 25000/26605
finished comment 26000/26605
33440
1
33440
10897
10539
3005
1

In [ ]:
data_final.head(3)

id   subreddit                                               body  \
0  xsjqzy  askscience  Why do I poop after a glass or two of beer or ...   
1  xsju1p  askscience  If you heated water under immense pressure so ...   
2  xsjx0l  askscience  What is the evolutionary goal of nose elongati...   

          author  score  gilded  created_utc  retreived_on  \
0  depressedchiq      1       0   1664591533    1665426496   
1   zhongliabuse      1       0   1664591793    1665426492   
2         _ozeki      1       0   1664592050    1665426489   

                                           permalink num_comments  \
0  /r/askscience/comments/xsjqzy/why_do_i_poop_af...            0   
1  /r/askscience/comments/xsju1p/if_you_heated_wa...            1   
2  /r/askscience/comments/xsjx0l/what_is_the_evol...            0   

                                                 url  self_text is_self  \
0  https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
1  https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
2  https://www.reddit.com/r/askscience/comments/x...  [removed]    True   

  parent_id link_id controversiality  is_submitter  
0    000000  000000          novalue          True  
1    000000  000000          novalue          True  
2    000000  000000          novalue          True

note: the 'id' column seems to be the comment id in teh comments data, whereas the 'parent_id' comment seems to be a link to the parent comment. In this data the parent id is not preceded by the 3 characters

need to remove the rows where body is 'removed' and author is 'deleted', or some combination of the two

In [ ]:
data_final.loc[data_final.author == '[deleted]', 'author'].count()

14490

In [ ]:
data_final.loc[data_final.author == '[removed]', 'author'].count()

0

In [ ]:
data_final.loc[data_final.body == '[deleted]', 'body'].count()

369

In [ ]:
data_final.loc[data_final.body == '[removed]', 'body'].count()

12926

these need to be removed

In [ ]:
data_final = data_final[data_final['body'] != '[removed]']
data_final = data_final[data_final['body'] != '[deleted]']
print(len(data_final))
print(len(pd.unique(data_final['author'])))
print(data_final.head(3))

20145
10896
       id   subreddit                                               body  \
0  xsjqzy  askscience  Why do I poop after a glass or two of beer or ...   
1  xsju1p  askscience  If you heated water under immense pressure so ...   
2  xsjx0l  askscience  What is the evolutionary goal of nose elongati...   

          author  score  gilded  created_utc  retreived_on  \
0  depressedchiq      1       0   1664591533    1665426496   
1   zhongliabuse      1       0   1664591793    1665426492   
2         _ozeki      1       0   1664592050    1665426489   

                                           permalink num_comments  \
0  /r/askscience/comments/xsjqzy/why_do_i_poop_af...            0   
1  /r/askscience/comments/xsju1p/if_you_heated_wa...            1   
2  /r/askscience/comments/xsjx0l/what_is_the_evol...            0   

                                                 url  self_text is_self  \
0  https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
1  http

In [ ]:
data_final = data_final[data_final['author'] != '[removed]']
data_final = data_final[data_final['author'] != '[deleted]']
print(len(data_final))
print(len(pd.unique(data_final['author'])))
print(data_final.head(3))

18949
10895
       id   subreddit                                               body  \
0  xsjqzy  askscience  Why do I poop after a glass or two of beer or ...   
1  xsju1p  askscience  If you heated water under immense pressure so ...   
2  xsjx0l  askscience  What is the evolutionary goal of nose elongati...   

          author  score  gilded  created_utc  retreived_on  \
0  depressedchiq      1       0   1664591533    1665426496   
1   zhongliabuse      1       0   1664591793    1665426492   
2         _ozeki      1       0   1664592050    1665426489   

                                           permalink num_comments  \
0  /r/askscience/comments/xsjqzy/why_do_i_poop_af...            0   
1  /r/askscience/comments/xsju1p/if_you_heated_wa...            1   
2  /r/askscience/comments/xsjx0l/what_is_the_evol...            0   

                                                 url  self_text is_self  \
0  https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
1  http

confirm that there are no more of the incorrect rows

In [ ]:
print(data_final.loc[data_final.author == '[deleted]', 'author'].count())
print(data_final.loc[data_final.author == '[removed]', 'author'].count())
print(data_final.loc[data_final.body == '[deleted]', 'body'].count())
print(data_final.loc[data_final.body == '[removed]', 'body'].count())

0
0
0
0


In [ ]:
data_final.head(10)

id   subreddit                                               body  \
0   xsjqzy  askscience  Why do I poop after a glass or two of beer or ...   
1   xsju1p  askscience  If you heated water under immense pressure so ...   
2   xsjx0l  askscience  What is the evolutionary goal of nose elongati...   
4   xsofdj  askscience  would we theoretically be able to see earths p...   
5   xsk9nb  askscience  How is the Calorie count of non-food items, su...   
6   xsotzf  askscience  Does lava loose temperature during it flow? If...   
8   xsox5d  askscience         Why do EVs use a bunch of small batteries?   
9   xsoxf5  askscience  Will solar panels wrapped around a tower gener...   
10  xskfqz  askscience  Is it possible to detect the microwaves suspec...   
11  xspaz8  askscience  CAN I GET A PARASITE BY ACCIDENTALLY EATING A ...   

                 author  score  gilded  created_utc  retreived_on  \
0         depressedchiq      1       0   1664591533    1665426496   
1          zhongliabuse      1       0   1664591793    1665426492   
2                _ozeki      1       0   1664592050    1665426489   
4          soggy-tampon      1       0   1664606931    1668049257   
5            liudinglue      1       0   1664593120    1665426473   
6   giftsignificant5286      1       0   1664608438    1668049244   
8             kenwillis      1       0   1664608764    1668049241   
9               phaiyez      1       0   1664608793    1668049241   
10            redzeusky      1       0   1664593609    1665426465   
11  sweet_disposition19      1       0   1664610193    1668049228   

                                            permalink num_comments  \
0   /r/askscience/comments/xsjqzy/why_do_i_poop_af...            0   
1   /r/askscience/comments/xsju1p/if_you_heated_wa...            1   
2   /r/askscience/comments/xsjx0l/what_is_the_evol...            0   
4   /r/askscience/comments/xsofdj/would_we_theoret...            1   
5   /r/askscience/comments/xsk9nb/how_is_the_calor...            0   
6   /r/askscience/comments/xsotzf/does_lava_loose_...            0   
8   /r/askscience/comments/xsox5d/why_do_evs_use_a...            0   
9   /r/askscience/comments/xsoxf5/will_solar_panel...            1   
10  /r/askscience/comments/xskfqz/is_it_possible_t...            0   
11  /r/askscience/comments/xspaz8/can_i_get_a_para...            1   

                                                  url  self_text is_self  \
0   https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
1   https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
2   https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
4   https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
5   https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
6   https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
8   https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
9   https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
10  https://www.reddit.com/r/askscience/comments/x...  [removed]    True   
11  https://www.reddit.com/r/askscience/comments/x...               True   

   parent_id link_id controversiality  is_submitter  
0     000000  000000          novalue          True  
1     000000  000000          novalue          True  
2     000000  000000          novalue          True  
4     000000  000000          novalue          True  
5     000000  000000          novalue          True  
6     000000  000000          novalue          True  
8     000000  000000          novalue          True  
9     000000  000000          novalue          True  
10    000000  000000          novalue          True  
11    000000  000000          novalue          True

In [ ]:
data_final.to_csv('/content/gdrive/MyDrive/Colab Notebooks/askscience/data_merged_askscience.csv',index=False)

how many rows are comments

In [ ]:
data_final.loc[data_final.is_submitter == False, 'is_submitter'].count()

12918

how many rows are submissions

In [ ]:
data_final.loc[data_final.is_submitter == True, 'is_submitter'].count()

6031

12918 + 6031 = 18949